aiperf/workers/worker.py

```diff
@@ -147,6 +163,246 @@ async def _handle_profile_cancel_command(
         )
         await self.stop()
 
+    async def _process_credit_drop_internal(self, message: CreditDropMessage) -> None:
+        """Process a credit drop message. Make sure to return the credit as soon as possible.
+
+        - Every credit must be returned after processing
+        - All results or errors should be converted to a RequestRecord and pushed to the inference results client.
+
+        NOTE: This function MUST NOT return until the credit drop is fully processed.
+        This is to ensure that the max concurrency is respected via the semaphore of the pull client.
+        The way this is enforced is by requiring that this method returns a CreditReturnMessage.
+        """
+        if self.is_trace_enabled:
+            self.trace(f"Processing credit drop: {message}")
+
+        self.task_stats.total += 1
```

Comment:
We may have to decide if we want this to represent conversation (credits) or requests. Right now you are increasing it once, but calling `task_finished` each response, which increases the done count, which will grow higher than the total.